src/SonarQube.TeamBuild.Integration/CoverageReportProcessorBase.cs

```diff
@@ -68,55 +67,50 @@ public bool ProcessCoverageReports()
 
             var success = TryGetBinaryReportFile(config, settings, logger, out string binaryFilePath);
 
-            if (success && binaryFilePath != null)
+            if (success &&
+                binaryFilePath != null &&
+                TryConvertCoverageReport(binaryFilePath, out var coverageReportPath) &&
+                !string.IsNullOrEmpty(coverageReportPath) &&
+                !config.LocalSettings.Any(IsVsCoverageXmlReportsPaths))
+            {
+                config.LocalSettings.Add( new Property { Id = SonarProperties.VsCoverageXmlReportsPaths, Value = coverageReportPath });
+            }
+
+            if (TryGetTrxFile(config, settings, logger, out var trxPath) &&
```

Comment:
Minor optimisation: no point in us accessing the disc to look for the file if the user has supplied the setting.